stdlib/REPL/src/REPL.jl

```diff
@@ -208,30 +208,31 @@ end
 
 ==(a::REPLDisplay, b::REPLDisplay) = a.repl === b.repl
 
-function display(d::REPLDisplay, mime::MIME"text/plain", x)
+function display(d::REPLDisplay, mime::MIME"text/plain", @nospecialize(x))
+    x = Ref{Any}(x)
     with_repl_linfo(d.repl) do io
         io = IOContext(io, :limit => true, :module => Main::Module)
         get(io, :color, false) && write(io, answer_color(d.repl))
         if isdefined(d.repl, :options) && isdefined(d.repl.options, :iocontext)
             # this can override the :limit property set initially
             io = foldl(IOContext, d.repl.options.iocontext, init=io)
         end
-        show(io, mime, x)
+        show(io, mime, x[])
```

Comment:
Nice trick! this is to compile only one specialization of the closure, and then the call to `show` will be a dymaic dispatch, correct?

Reply:
Yes.

Reply:
Though we should really handle this by propagating `nospecialize` to closure types.